search tool: show also in other desktops

genodeftest opened the pull request:
With GNOME search tool being deprecated and removed from some distros therefore, how about making this app available in other desktops too?

lukefromdc reviewed (approved):
This will be useful, and I don't see any way removing this one line from the .desktop file could cause a problem. Approving and will merge

raveit65 commented:
> This will be useful, and I don't see any way removing this one line from the .desktop file could cause a problem

Looks like you want to work on bugreports in gnome or other desktops.
Thank you for your single man power :-)

lukefromdc commented:
Do you want to revert this? I am OK with that, was only thinking of the app being more widely useful.

Note that I have had to try to keep things afloat singlehandedly most of the summer, but cannot fix
the current round of Travis problems. I've had no guidance on how to handle this situation and honestly
am in over my head at this point

lukefromdc commented:
Also, note that no MATE wayland application that uses any feature of GtkLayerShell or Wlroots will be able 
to run properly in a GNOME wayland session. Menus and popups will go to the center of the screen at least,
or apps might crash entirely. 

Thus any reports of that sort of bugs will have to be closed, as I have no intention of adding code to fix
gnome-shell wayland issues.

This one should certainly run OK in a GNOME x11 session and any wlroots wayland session, but with no 
guarantees. Given that, should I revert this and more broadly ensure MATE apps don't appear by default
in anyone else's menus? Alternately, smaller DE's often do use MATE components notably Pluma as the
GNOME versions either cannot be made to match them, or (gedit) have gone unsupported. This makes a
case for not limiting MATE apps to MATE sessions, though I don't see any of us being able to debug issues
that show up only in sessions we don't have installed.

raveit65 commented:
In fedora workstation (gnome 4) you only have to type in search field to get installed application. No idea why i should use mate-search-tool there.
From my distro maintainer view this PR will only cause more unneeded responsibility and work.
I only received a mail notification  about this PR, but honestly i don't wanna step in again.
Maybe ask some other old (spanish) Mate developer for help, maybe the will come back after germ bad ass is gone :)

raveit65 commented:
PS, you can use `OnlyShowIn= Mate, xfce, <or any other desktop>` , to allow other small desktops to use Mate applications.

lukefromdc commented:
I have reverted this commit. Given the current team status we cannot service 
"doesn't work in (some other DE)" bug reports. We cannot take on any more 
work at this point unless new folks join the team.

genodeftest commented:
> Also, note that no MATE wayland application that uses any feature of GtkLayerShell or Wlroots will be able to run properly in a GNOME wayland session. Menus and popups will go to the center of the screen at least, or apps might crash entirely.

I am using MATE search tool 1.28.0 on GNOME wayland session (GNOME shell 46.5) and this is not correct in general. As far as I can see, dropdown menus and context menus are placed correctly in MATE search tool. Maybe it breaks only on HiDPI screens or other contexts, but it works for me.

> In fedora workstation (gnome 4) you only have to type in search field to get installed application. No idea why i should use mate-search-tool there.

Those are very different use cases. Mate search tool allows exact search, filtering by date, exact search and maybe others – none of which GNOME shell's overview search can fullfil. Except for performance, an index-based search will always be worse for many use cases than a full text search. Actually, it was my motivation to have MATE search tool on GNOME because of this. (And yes, there was GNOME search tool, but it seems like it has been discontinued, as the last source release was 12 years ago and I cannot find it on GNOME's gitlab. Fedora has removed the package almost 2 years ago.)

As a workaround, I have copied the .desktop file from `/usr/share/applications` to `~/.local/share/applications` and removed the restriction to the MATE desktop. After reading your comments, I understand that I cannot expect any support for this unintended usage.

> I have reverted this commit. Given the current team status we cannot service
"doesn't work in (some other DE)" bug reports.

Understood. Sorry for the noise.

lukefromdc commented:
I don't think mate-search-tool uses and GtkLayerShell or Wlroots specific code, which would explain why this one
is working in gnome-shell. Mate panel or caja's desktop window would not.
